compiler/optimizer/VPHandlersCommon.cpp

```diff
@@ -69,8 +70,25 @@ TR::Node *constrainChildren(OMR::ValuePropagation *vp, TR::Node *node)
    TR::Node *myParent = vp->getCurrentParent();
    for (int32_t i = node->getNumChildren()-1; i >= 0; i--)
       {
+      TR::Node* child = node->getChild(i);
       vp->setCurrentParent(node);
-      vp->launchNode(node->getChild(i), node, i);
+      vp->launchNode(child, node, i);
+
+      // Replace PassThrough with its child
+      if (child->getOpCodeValue() == TR::PassThrough
+          && !node->getOpCode().isNullCheck())
+         {
+         TR::Node* grandChild = child->getFirstChild();
+         node->setAndIncChild(i, grandChild);
+
+         if (child->getReferenceCount() == 1)
+            child->recursivelyDecReferenceCount();
+         else
+            child->decReferenceCount();
```

Comment:
Couldn't this conditional be replaced with an unconditional call to `recursivelyDecReferenceCount()`?